src/components/ConfigureSupersetEmbeddedDashboardModal/UpdateSupersetEmbeddedDashboard.js

```diff
@@ -0,0 +1,219 @@
+                </ModalTitle>
+                <ModalContent className={styles.modalContent}>
+                    {(queryLoading || !!queryError) && (
+                        <div
+                            className={cx(styles.contentOverlay, {
+                                [styles.loading]: queryLoading,
+                                [styles.error]: queryError,
+                            })}
+                        >
+                            {queryLoading && <CircularLoader />}
+                            {queryError && (
+                                <NoticeBox
+                                    error
+                                    title={i18n.t(
+                                        'Could not load dashboard details'
+                                    )}
+                                >
+                                    {parseErrorText(queryError)}
+                                </NoticeBox>
+                            )}
+                        </div>
+                    )}
+                    <SupersetEmbeddedDashboardFields
+                        isSupersetEmbedIdValid={isSupersetEmbedIdValid}
+                        isSupersetEmbedIdFieldTouched={
+                            isSupersetEmbedIdFieldTouched
+                        }
+                        values={values}
+                        onChange={onChange}
+                        onSupersetEmbedIdFieldBlur={onSupersetEmbedIdFieldBlur}
+                        submitting={mutationLoading || queryLoading}
+                    />
+                    {mutationError && (
+                        <NoticeBox error title={mutationErrorTitle}>
+                            {mutationErrorText}
+                        </NoticeBox>
+                    )}
+                </ModalContent>
+                <ModalActions>
+                    <div className={styles.buttonStrip}>
+                        {!queryError && (
+                            <Button
+                                loading={mutationLoading}
+                                type="submit"
+                                primary
+                                disabled={
+                                    !hasFieldChanges ||
+                                    !isSupersetEmbedIdValid ||
+                                    queryLoading
+                                }
+                            >
+                                {i18n.t('Update dashboard')}
+                            </Button>
+                        )}
+                        <Button
+                            disabled={mutationLoading}
+                            secondary={!queryError}
+                            primary={!!queryError}
+                            onClick={closeModal}
+                            type={queryError ? 'submit' : undefined}
+                        >
+                            {i18n.t('Cancel')}
+                        </Button>
```

Comment:
When the dashboard-details fail to load, the "Cancel" button becomes the primary submit button, so a user can hit ENTER to close.